ci/tests/vmachine/vmachine_test.py

```diff
@@ -77,6 +82,11 @@ def setup():
 
 def teardown():
     api = Connection.get_connection()
+    template_machine_list = api.get_component_by_name('vmachines', TEMPLATE_MACHINE_NAME)
+    if template_machine_list and len(template_machine_list):
+        VMachineController.delete(template_machine_list[0]['guid'])
```

Comment:
You only delete the 1st machine from the list, but you do expect all machines with that name to be gone afterwards?